lib/ansible/modules/cloud/google/gcp_forwarding_rule.py

```diff
@@ -0,0 +1,378 @@
+                 initial forwarding_rule creation.
+    returned: if the forwarding_rule has been updated.
+    type: bool
+    sample: true
+'''
+
+
+try:
+    from ast import literal_eval
+    HAS_PYTHON26 = True
+except ImportError:
+    HAS_PYTHON26 = False
+
+# import module snippets
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.gcp import get_google_api_client, GCPUtils
+
+USER_AGENT_PRODUCT = 'ansible-forwarding_rule'
+USER_AGENT_VERSION = '0.0.1'
+
+
+def _build_global_forwarding_rule_dict(params, project_id=None):
+    """
+    Reformat services in Ansible Params.
+
+    :param params: Params from AnsibleModule object
+    :type params: ``dict``
+
+    :param project_id: The GCP project ID.
+    :type project_id:  ``str``
+
+    :return: dictionary suitable for submission to GCP API.
+    :rtype ``dict``
+    """
+    url = ''
+    if project_id:
+        url = GCPUtils.build_googleapi_url(project_id)
+    gcp_dict = GCPUtils.params_to_gcp_dict(params, 'forwarding_rule_name')
+    if 'target' in gcp_dict:
+        gcp_dict['target'] = '%s/global/targetHttpProxies/%s' % (url,
+                                                                 gcp_dict['target'])
+    if 'address' in gcp_dict:
+        gcp_dict['IPAddress'] = '%s/global/addresses/%s' % (url,
+                                                            gcp_dict['address'])
+        del gcp_dict['address']
+    if 'protocol' in gcp_dict:
+        gcp_dict['IPProtocol'] = gcp_dict['protocol']
+        del gcp_dict['protocol']
+    return gcp_dict
+
+
+def get_global_forwarding_rule(client, name, project_id=None):
+    """
+    Get a Global Forwarding Rule from GCP.
+
+    :param client: An initialized GCE Compute Disovery resource.
+    :type client:  :class: `googleapiclient.discovery.Resource`
+
+    :param name: Name of the Global Forwarding Rule.
+    :type name:  ``str``
+
+    :param project_id: The GCP project ID.
+    :type project_id:  ``str``
+
+    :return: A dict resp from the respective GCP 'get' request.
+    :rtype: ``dict``
+    """
+    try:
+        req = client.globalForwardingRules().get(
+            project=project_id, forwardingRule=name)
+        return GCPUtils.execute_api_client_req(req, raise_404=False)
+    except:
+        raise
+
+
+def create_global_forwarding_rule(client, params, project_id):
+    """
+    Create a new Global Forwarding Rule.
+
+    :param client: An initialized GCE Compute Disovery resource.
+    :type client:  :class: `googleapiclient.discovery.Resource`
+
+    :param params: Dictionary of arguments from AnsibleModule.
+    :type params:  ``dict``
+
+    :return: Tuple with changed status and response dict
+    :rtype: ``tuple`` in the format of (bool, dict)
+    """
+    gcp_dict = _build_global_forwarding_rule_dict(params, project_id)
+    with open('/tmp/mylog.log', 'w') as f:
+        f.write(str(params))
+        f.write(str(gcp_dict))
+    f.close()
```

Comment:
the `f.close()` is implied as part of the with block. Also, I don't think this should be in the module since it looks like a debug statement.